RULES_DESCRIPTIONS.md

```diff
@@ -892,6 +893,11 @@ Example:
 [rule.superfluous-else]
   arguments = ["preserveScope"]
 ```
+## time-after-leak
+
+_Description_: This rule warns on temporary goroutine leak when using `time.After` in select statements. In Go versions before 1.23, [the garbage collector does not recover the underlying thread until the timer fires](https://pkg.go.dev/time#After).
+
+_Configuration_: N/A
```

Comment:
I propose adding a note similar to the ones in `range-val-address` and `range-val-in-closure`:
```suggestion
_Configuration_: N/A

_Note_: This rule is irrelevant for Go 1.23+.
```